add-on/src/lib/ipfs-client/index.js

```diff
@@ -55,46 +56,32 @@ async function destroyIpfsClient (browser) {
   if (!client) return
   try {
     await client.destroy(browser)
-    await _reloadIpfsClientDependents(browser) // sync (API stopped working)
+    await reloadIpfsClientDependents(browser) // sync (API stopped working)
   } finally {
     client = null
   }
 }
 
-function _isWebuiTab (url) {
-  const bundled = !url.startsWith('http') && url.includes('/webui/index.html#/')
-  const ipns = url.includes('/webui.ipfs.io/#/')
-  return bundled || ipns
-}
-
-function _isInternalTab (url, extensionOrigin) {
-  return url.startsWith(extensionOrigin)
-}
-
-async function _reloadIpfsClientDependents (browser, instance, opts) {
+async function reloadIpfsClientDependents (
+  browser, instance, opts, reloadExtensions = [WebUiReloader, LocalGatewayReloader, InternalTabReloader]) {
   // online || offline
   if (browser.tabs && browser.tabs.query) {
     const tabs = await browser.tabs.query({})
     if (tabs) {
-      const extensionOrigin = browser.runtime.getURL('/')
-      tabs.forEach((tab) => {
-        // detect bundled webui in any of open tabs
-        if (_isWebuiTab(tab.url)) {
-          log(`reloading webui at ${tab.url}`)
-          browser.tabs.reload(tab.id)
-        } else if (_isInternalTab(tab.url, extensionOrigin)) {
-          log(`reloading internal extension page at ${tab.url}`)
-          browser.tabs.reload(tab.id)
-        }
-      })
+      const reloadChecks = await prepareReloadExtensions(reloadExtensions, browser, log)
+      reloadChecks.forEach(check => check.reload(tabs))
```

Comment:
extensions should be async (if they're not already). So we should not use a forEach to loop over a reload operation.

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach#sect2 and https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach#sect3 for reasons why

Reply:
If we don't want to `await` each reload sequentially, that's fine too, but we should still wrap in a try/catch so we don't have floating promises.

Reply:
Added the try-catch block, it won't catch the promises inside the `forEach` block though, unless we explicitly await. I wanted this to be fire and forget.